src/app/gps/qgsappgpstools.cpp

```diff
@@ -0,0 +1,35 @@
+/***************************************************************************
+    qgsappgpstools.cpp
+    ---------------------
+    begin                : May 2025
+    copyright            : (C) 2025 by Germán Carrillo
+    email                : german at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgisapp.h"
+#include "qgsappgpstools.h"
+
+QgsAppGpsTools::QgsAppGpsTools()
+{
+}
+
+void QgsAppGpsTools::setGpsPanelConnection( QgsGpsConnection *connection )
+{
+  QgisApp::instance()->setGpsPanelConnection( connection );
+}
+
+void QgsAppGpsTools::createFeatureFromGpsTrack()
+{
+  QgisApp::instance()->createFeatureFromGpsTrack();
+}
+
+void QgsAppGpsTools::setGpsTrackLineSymbol( QgsLineSymbol *symbol )
+{
+  QgisApp::instance()->setGpsTrackLineSymbol( symbol );
```

Comment:
Could we call QgsAppGpsDigitizing directly here, avoiding the need for new methods in QgisApp?

```suggestion
  QgsAppGpsDigitizing::setGpsTrackLineSymbol( symbol );
```

Reply:
That would be good indeed, but we'd still need to call:

`mGpsDigitizing->updateTrackAppearance();`

Which is non-static. Same for

`mGpsDigitizing->createFeature();`

in the method above.

What would be your suggestion, could we expose `mGpsDigitizing` from `QgisApp` instead of the new methods?

Reply:
Yes, that sounds like a good idea!